Lusas_Adapter/CRUD/Create/Elements/Surface.cs

```diff
@@ -108,6 +144,44 @@ private IFSurface CreateSurface(Panel panel)
 
         /***************************************************/
 
+        private IFSurface CreateSurface(Opening opening)
+        {
+            List<IFLine> edges = new List<IFLine>();
+
+            foreach (Edge edge in opening.Edges)
+            {
+                string edgeId = GetAdapterId<string>(edge);
+
+                if (string.IsNullOrEmpty(edgeId))
+                {
+                    Engine.Base.Compute.RecordError("Could not find the ids for at least one Edge, Panel not created.");
+                    return null;
+                }
+                else
+                {
+                    edges.Add(d_LusasData.getLineByNumber(edgeId));
+                }
+            }
+
+            IFSurface lusasSurface = d_LusasData.createSurfaceBy(edges.ToArray());
+
+            if (lusasSurface != null) //Is this neccesary ?
```

Comment:
```suggestion
            if (lusasSurface != null)
```
It just guards against a scenario we have not accounted for, where `lusasSurface = null` because then you will get a load of exceptions downstream because the `IFSurface` is null. For example, if you query the `Id` that will return an error and the code will break. This avoids that.